### ISSUE TITLE
Read No GPU device on MacBook Air M1

### ISSUE THREAD

**Issue by BenjaminChoou** [apple/tensorflow_macos#112]
I installed the latest release tf2.4 for mac on my M1 MacBook. And I test the `tf.config.list_physical_devices()` to check whether GPU is involved. What I got is '[PhysicalDevice(name='/physical_device:CPU:0', device_type='CPU')]'. Could anyone explain it why there is no GPU device? Does tf2.4 utilize the GPU device on M1 chip?

**Comment by iRonJ** [comment]
It uses MLCompute which I'm assuming should use the neural engine, but the Apple GPU isn't a compute device.

**Comment by BenjaminChoou** [comment]
> ute device.

Yeah! I think you're right.

**Comment by BenjaminChoou** [comment]
> It uses MLCompute which I'm assuming should use the neural engine, but the Apple GPU isn't a compute device.

Yeah! I think you're right. But it seems there is no way to monitor the usage of neural engine.

**Comment by BenjaminChoou** [comment]
@iRonJ 

> It uses MLCompute which I'm assuming should use the neural engine, but the Apple GPU isn't a compute device.

Yeah! I think you're right. But it seems there is no way to monitor the usage of neural engine.

**Comment by anna-tikhonova** [comment]
Currently, we not have support for the tf.config.list_physical_devices() API. There is an optional mlcompute.set_mlc_device(device_name=’any') API for ML Compute device selection. The default value for device_name is 'any’, which means ML Compute will select the best available device on your system, including multiple GPUs on multi-GPU configurations. Other available options are ‘cpu’ and ‘gpu’. Please see the Device Selection and the Logging and Debugging sections of the README for more information. If you turn on logging, you can see which device(s) ML Compute is using.

**Comment by BenjaminChoou** [comment]
> Currently, we not have support for the tf.config.list_physical_devices() API. There is an optional mlcompute.set_mlc_device(device_name=’any') API for ML Compute device selection. The default value for device_name is 'any’, which means ML Compute will select the best available device on your system, including multiple GPUs on multi-GPU configurations. Other available options are ‘cpu’ and ‘gpu’. Please see the Device Selection and the Logging and Debugging sections of the README for more information. If you turn on logging, you can see which device(s) ML Compute is using.

Thank you for your explanation.

**Comment by watjurk** [comment]
It seems that this issue should be marked as closed @BenjaminChoou @anna-tikhonova.

Thanks!

**Comment by ManuelSchneid3r** [comment]
@anna-tikhonova please put this under the notes in the readme. This took me about a week to figure out.
